addons/website_forum/models/ir_attachment.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class Attachment(models.Model):
+
+    _inherit = "ir.attachment"
+
+    def _can_bypass_rights_on_media_dialog(self, **attachment_data):
+        # Bypass the attachment create ACL and let the user create the image
+        # attachment if he has write right on the model (the image attachment
+        # will be bound to this model's record).
+        res_model = attachment_data['res_model']
+        res_id = attachment_data.get('res_id')
+        if (
+            res_model == 'forum.post' and res_id
+            and self.env['forum.post'].browse(res_id).can_comment
```

Comment:
The real condition should be related to `karma_editor` fields (`can_comment` is too soft) but unfortunately, there is computed field for that yet... worth adding now?

Reply:
I am not sure this will go through, but added it with https://github.com/odoo/odoo/compare/ea5ac1a5f970b04e969818b9a18526ab857f9051..3b5fad416ef914736523b7e97ca1ae9404de42a2
I believe we will need an exception